src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String DATE_SEPARATOR = " - ";
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int RATE_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ").append(dateFrom).append(DATE_SEPARATOR).append(dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (String item : data) {
+                String[] parts = item.split(" ");
+                if (isWithinRange(parts[DATE_INDEX], dateFrom, dateTo)
+                        && name.equals(parts[NAME_INDEX])) {
+                    int hours = Integer.parseInt(parts[HOURS_INDEX]);
+                    int rate = Integer.parseInt(parts[RATE_INDEX]);
+                    salary += hours * rate;
+                }
+            }
+            report.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+        }
+        return report.toString();
+    }
+
+    private static boolean isWithinRange(String date, String dateFrom, String dateTo) {
+        try {
```

Comment:
is date parsing return any checked exception? Do you really need try-catch here?